### PR TITLE
Removed check for ensemble when training best pipeline

### LINKED ISSUE #1931: AutoML: Train best pipeline on entire dataset if it's an ensemble
Right now when we go to train the best pipeline after search and the pipeline is an ensemble, [we train the best pipeline on the separate ensemble split](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_search.py#L593):

`self._best_pipeline.fit(X_train.iloc[ensemble_indices], y_train.iloc[ensemble_indices])`

After discussing this with @dsherry @rpeck @bchen1116 and @angela97lin, we agreed it's valid and preferred to train the ensemble on the entire dataset, i.e `self._best_pipeline.fit(X_train, y_train)`

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes #1931

### CODE REVIEW

**Comment by ParthivNaresh** [comment]
Perf tests: https://alteryx.quip.com/ZXV4AcYh5eVc/Best-Pipeline-Ensemble-training-on-whole-dataset-Performance-Tests

File: evalml/automl/engine/sequential_engine.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def evaluate_batch(self, pipelines):
             X, y = self.X_train, self.y_train
             if pipeline.model_family == ModelFamily.ENSEMBLE:
                 X, y = self.X_train.iloc[self.ensembling_indices], self.y_train.iloc[self.ensembling_indices]
-            elif self.ensembling_indices is not None:
+            elif self.ensembling_indices is not None:   # Is this necessary?
```
Comment (chukarsten) [inline evalml/automl/engine/sequential_engine.py:37]: Someone will have to comment on this.  When this got extracted from AutoML into the engine, we lost track of who put it in there.  It's kind of strange, undocumented behavior.

**Review by chukarsten — APPROVED** [review]
This looks good.  I think it would be good to answer the question you pointed out in the sequential engine, but that's not blocking your work.

**Comment by ParthivNaresh** [comment]
I'm turning this back into a draft PR because I can't confirm that Ensembling is running for these performance tests. I need at least 50 iterations and unfortunately every job I've tried to run with it hasn't finished. I'll be looking for another way around this.

**Comment by ParthivNaresh** [comment]
I've decided to run with limited model families to reduce the iterations, this should result in a run while still allowing ensembling to take place

**Review by bchen1116 — APPROVED** [review]
LGTM! Thanks for making these changes. Left one cleanup comment.

File: evalml/tests/automl_tests/test_automl.py [hunk 2]
```diff
@@ -2254,15 +2254,13 @@ def test_automl_ensembling_best_pipeline(mock_fit, mock_score, mock_rankings, in
...
-    training_indices, ensembling_indices, _, _ = split_data(ww.DataTable(np.arange(X.shape[0])), y, problem_type='binary', test_size=ensemble_split_size, random_seed=0)
-    training_indices, ensembling_indices = training_indices.to_dataframe()[0].tolist(), ensembling_indices.to_dataframe()[0].tolist()
     # when best_pipeline == -1, model is ensembling,
     # otherwise, the model is a different model
     # the ensembling_num formula is taken from AutoMLSearch
     if best_pipeline == -1:
         assert automl.best_pipeline.model_family == ModelFamily.ENSEMBLE
-        assert len(mock_fit.call_args_list[-1][0][0]) == len(ensembling_indices)
-        assert len(mock_fit.call_args_list[-1][0][1]) == len(ensembling_indices)
+        assert len(mock_fit.call_args_list[-1][0][0]) == len(X)
```
[the remark this answers is not in the document]
Comment (ParthivNaresh) [reply evalml/tests/automl_tests/test_automl.py]: Great point!

File: evalml/automl/automl_search.py [hunk 3]
```diff
@@ -529,11 +529,8 @@ def _find_best_pipeline(self):
         if not (self._best_pipeline and self._best_pipeline == self.get_pipeline(best_pipeline['id'])):
             best_pipeline = self.get_pipeline(best_pipeline['id'])
             if self._train_best_pipeline:
-                if best_pipeline.model_family == ModelFamily.ENSEMBLE:
-                    X_train, y_train = self.X_train.iloc[self.ensembling_indices], self.y_train.iloc[self.ensembling_indices]
-                else:
-                    X_train = self.X_train
-                    y_train = self.y_train
+                X_train = self.X_train
+                y_train = self.y_train
```
Comment (dsherry) [inline evalml/automl/automl_search.py:533]: 👏  beautiful

**Review by dsherry — APPROVED** [review]
🚢 !

**Pull request merged**